api/models/Notification.js

```diff
@@ -1,31 +1,246 @@
 
-  actor: function() {
-    return this.belongsTo(User);
+  tableName: 'notifications',
+
+  activity: function () {
+    return this.belongsTo(Activity)
+  },
+
+  send: function () {
+    var action
+    switch (this.get('medium')) {
+      case Notification.MEDIUM.Push:
+        action = this.sendPush()
+        break
+      case Notification.MEDIUM.Email:
+        action = this.sendEmail()
+        break
+    }
+    if (action) {
+      return action
+      .then(() => this.save({'sent_at': (new Date()).toISOString()}))
+    } else {
+      return Promise.resolve()
+    }
+  },
+
+  sendPush: function () {
+    switch (Notification.priorityReason(this.relations.activity.get('meta').reasons)) {
+      case 'mention':
+        return this.sendPostPush('mention')
+      case 'commentMention':
+        return this.sendCommentPush('mention')
+      case 'newComment':
+        return this.sendCommentPush()
+      case 'tag':
+        return Promise.resolve()
+      case 'newPost':
+        return this.sendPostPush()
+      default:
+        return Promise.resolve()
+    }
   },
 
-  post: function() {
-    return this.belongsTo(Post);
+  sendPostPush: function (version) {
+    var post = this.relations.activity.relations.post
+    var communityIds = Activity.communityIds(this.relations.activity)
+    if (isEmpty(communityIds)) return Promise.resolve()
+    return Community.find(communityIds[0])
+    .then(community => {
+      var path = url.parse(Frontend.Route.post(post, community)).path
+      var alertText = PushNotification.textForPost(post, community, this.relations.activity.get('reader_id'), version)
+      return this.relations.activity.relations.reader.sendPushNotification(alertText, path)
+    })
   },
 
-  action: function() {
-    return this.get('action');
+  sendCommentPush: function (version) {
+    var comment = this.relations.activity.relations.comment
+    var communityIds = Activity.communityIds(this.relations.activity)
+    if (isEmpty(communityIds)) return Promise.resolve()
+    return Community.find(communityIds[0])
+    .then(community => {
+      var path = url.parse(Frontend.Route.post(comment.relations.post, community)).path
+      var alertText = PushNotification.textForComment(comment, version, this.relations.activity.get('reader_id'))
+      return this.relations.activity.relations.reader.sendPushNotification(alertText, path)
+    })
   },
 
-  timestamp: function() {
-    return this.get('timestamp');
+  sendEmail: function () {
+    switch (Notification.priorityReason(this.relations.activity.get('meta').reasons)) {
+      case 'mention':
+        return this.sendPostMentionEmail()
+      case 'commentMention':
+        return this.sendCommentNotificationEmail('mention')
+      case 'newComment':
+        return this.sendCommentNotificationEmail()
+      default:
+        return Promise.resolve()
+    }
+  },
+
+  sendPostMentionEmail: function () {
+    var post = this.relations.activity.relations.post
```

Comment:
maybe make some instance methods just to reduce the monotony of seeing `relations` over and over, e.g.
```js
post: function () {
  return this.relations.activity.relations.post
}
```